### PR TITLE
Secure API request via Mattermost-User-ID header

### PR DESCRIPTION
**Pull request by hanzei**

### CODE REVIEW

File: server/plugin/api.go [hunk 1]
```diff
@@ -116,7 +117,18 @@ func (p *MatterpollPlugin) handleLogo(w http.ResponseWriter, r *http.Request) {
...
 
-func (p *MatterpollPlugin) handlePostActionIntegrationRequest(handler postActionHandler) func(http.ResponseWriter, *http.Request) {
+func checkAuthenticity(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Header.Get("Mattermost-User-ID") == "" {
+			http.Error(w, "not authorized", http.StatusUnauthorized)
+			return
+		}
+
+		next.ServeHTTP(w, r)
```
Comment (hanzei) [inline server/plugin/api.go]: I love this solution.

File: server/plugin/api_test.go [hunk 2]
```diff
@@ -632,7 +680,8 @@ func TestHandleEndPoll(t *testing.T) {
 			p := setupTestPlugin(t, api, store)
 
 			w := httptest.NewRecorder()
-			r := httptest.NewRequest("POST", fmt.Sprintf("/api/v1/polls/%s/end", testutils.GetPollID()), bytes.NewReader(test.Request.ToJson()))
+			r := httptest.NewRequest(http.MethodPost, fmt.Sprintf("/api/v1/polls/%s/end", testutils.GetPollID()), bytes.NewReader(test.Request.ToJson()))
```
Comment (kaakaa) [inline server/plugin/api_test.go]: `TestHandleAddOptionDialogRequest` doesn't have a test for `not-authorized`.   Are there any reasons?

File: server/plugin/api.go [hunk 3]
```diff
@@ -85,13 +85,14 @@ func (p *MatterpollPlugin) InitAPI() *mux.Router {
 	r.HandleFunc("/"+iconFilename, p.handleLogo).Methods("GET")
```
Comment (kaakaa) [inline server/plugin/api.go]: Would you change `"GET"` to `http.MethodGet` for consistency?

File: server/plugin/api_test.go (hunk 2 above)
Comment (hanzei) [reply server/plugin/api_test.go]: Nope, I just forgot it. Added a test for this.

File: server/plugin/api.go (hunk 3 above)
Comment (hanzei) [reply server/plugin/api.go]: Good catch :+1:

**Pull request merged**